t/auth_jwt_require_header.t

```diff
@@ -0,0 +1,76 @@
+use Test::Nginx::Socket 'no_plan';
+
+no_root_location();
+no_shuffle();
+
+run_tests();
+
+__DATA__
+
+=== test: valid operators
+--- http_config
+include $TEST_NGINX_CONF_DIR/authorized_server.conf;
+--- config
+include $TEST_NGINX_CONF_DIR/jwt.conf;
+location / {
+  set $variable "1";
+  auth_jwt "" token=$test1_jwt;
+  auth_jwt_require_header exp eq $variable;
+  auth_jwt_require_header exp ne $variable;
+  auth_jwt_require_header exp gt $variable;
+  auth_jwt_require_header exp ge $variable;
+  auth_jwt_require_header exp lt $variable;
+  auth_jwt_require_header exp le $variable;
+  auth_jwt_require_header exp intersect $variable;
+  auth_jwt_require_header exp nintersect $variable;
+  auth_jwt_require_header exp in $variable;
+  auth_jwt_require_header exp nin $variable;
+}
+--- request
+    GET /
+--- error_code: 401
+
+
+=== test: check header requirements returns 200 for simple operator eq (others operators are the same like for jwt-claims)
+--- http_config
+include $TEST_NGINX_CONF_DIR/authorized_server.conf;
+--- config
+include $TEST_NGINX_CONF_DIR/jwt.conf;
+location / {
+  set $expected_kid "test1";
+  auth_jwt "" token=$test1_jwt;
+  auth_jwt_require_header kid eq $expected_kid;
+}
+--- request
+    GET /
+--- error_code: 401
+
+=== test: check 401 with invalid operators
+--- http_config
+include $TEST_NGINX_CONF_DIR/authorized_server.conf;
+--- config
+include $TEST_NGINX_CONF_DIR/jwt.conf;
+location / {
+  set $variable "1";
+  auth_jwt "" token=$test1_jwt;
+  auth_jwt_require_header exp invalidname $variable;
+}
+--- must_die
+
+=== test: check 401 with invalid json expected variable
+--- http_config
+include $TEST_NGINX_CONF_DIR/authorized_server.conf;
+--- config
+include $TEST_NGINX_CONF_DIR/jwt.conf;
+location / {
+  set $variable "'1'"; # it is not a json string
+  auth_jwt "" token=$test1_jwt;
+  auth_jwt_require_claim exp eq $variable;
+}
+--- request
+    GET /
+--- error_code: 401
+--- error_log
+auth_jwt: failed to json_load jwt claim requirement
```

Comment:
auth_jwt: failed to json_load jwt header requirement

Reply:
Fixed